torch/nn/modules/batchnorm.py

```diff
@@ -440,19 +437,24 @@ def forward(self, input):
             else:  # use exponential moving average
                 exponential_average_factor = self.momentum
 
-        world_size = 1
-        process_group = torch.distributed.group.WORLD
-        if self.process_group:
-            process_group = self.process_group
-        world_size = torch.distributed.get_world_size(process_group)
+        need_sync = self.training or not self.track_running_stats
+        if need_sync:
+            process_group = torch.distributed.group.WORLD
+            if self.process_group:
+                process_group = self.process_group
+            world_size = torch.distributed.get_world_size(process_group)
+            need_sync = need_sync and world_size > 1
```

Comment:
`need_sync = world_size > 1` should be sufficient, as otherwise, it wouldn't enter this clause.